### ISSUE TITLE
Burger menu causes crash under Wayland (Fedora 36)

### ISSUE THREAD

**Issue by jfmcbrayer** [skyjake/lagrange#502]
Lagrange version: 1.13.4 (release branch from git)

SDL version: SDL2-2.0.22

Wayland compositors tested: gnome-shell 42.0, Sway 1.7

On Fedora 36, wayland is the default driver for SDL2, even if it's not yet the default on other distributions. So either `./lagrange` or `env SDL_VIDEODRIVER=wayland ./lagrange` crash whenever you click on the burger menu. Or I should say, the window closes, but the program keeps running, unresponsive, in the background.
The Fedora-packaged lagrange (1.2.1) also crashes the same way. 

'env SDL_VIDEODRIVER=x11 ./lagrange` works fine.

Here's the output from stderr:

```
Lagrange: A Beautiful Gemini Client
[FontPack] 20 fonts available
[window] renderer: opengl (accelerated)
[Text] subpixel offsets: 4
[Text] 845 font variants ready
[window] max texture size: 16384 x 16384
[window] supported texture format: SDL_PIXELFORMAT_ARGB8888
[window] supported texture format: SDL_PIXELFORMAT_ABGR8888
[window] supported texture format: SDL_PIXELFORMAT_RGB888
[window] supported texture format: SDL_PIXELFORMAT_BGR888
[window] supported texture format: SDL_PIXELFORMAT_YV12
[window] supported texture format: SDL_PIXELFORMAT_IYUV
[window] supported texture format: SDL_PIXELFORMAT_NV12
[window] supported texture format: SDL_PIXELFORMAT_NV21
[Text] subpixel offsets: 4
[Text] subpixel offsets: 4
[Text] 845 font variants ready
[GmRequest] URL: gemini://gemini.circumlunar.space/
[Text] subpixel offsets: 4
[Text] 845 font variants ready
[Text] subpixel offsets: 4
[Text] 845 font variants ready
[window] renderer: software
[Text] subpixel offsets: 4
[Text] 845 font variants ready
[newPopup_Window] 11272 μs
wl_display@1: error 1: invalid arguments for xdg_surface@129.get_popup
```

I could probably produce a core dump (by sending SIGQUIT to the hung process), if it would help.

**Comment by jfmcbrayer** [comment]
After reading closed bug reports on this, I tried using the -DENABLE_POPUP_MENUS=NO to cmake. It now works correctly. If you want to close this, that's fine; I should probably report it to the Fedora packager.

**Comment by skyjake** [comment]
> wl_display@1: error 1: invalid arguments for xdg_surface@129.get_popup

It looks like SDL's Wayland support is still incomplete.

This is also mentioned in the "Known issues" section on the [homepage](https://gmi.skyjake.fi/lagrange/):
> The environment variable `SDL_VIDEODRIVER=wayland` can be used to force SDL to use Wayland instead of X11. However, SDL does not have full support for Wayland at the moment. You should set the Lagrange CMake build option ENABLE_POPUP_MENUS to NO. If there are issues with resizing windows, also set ENABLE_RESIZE_DRAW to NO.

**Comment by alexpdp7** [comment]
Reported as https://bugzilla.redhat.com/show_bug.cgi?id=2135123 .

**Comment by michel-slm** [comment]
Ah, I've not been noticing since
- I accidentally switched to X11 on my Fedora box a few months ago
- my Lagrange window is normally big enough the menu does not pop up

Can repro, will adjust the settings, thanks!

**Comment by michel-slm** [comment]
For those following along, this will be fixed in Fedora/EPEL together in the 1.14.2-2 update: https://bodhi.fedoraproject.org/updates/?search=1.14.2-2&packages=lagrange
